cogs/make_applicant.py

```diff
@@ -0,0 +1,192 @@
+            main_guild.text_channels,
+            name="introductions",
+        )
+
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Attempting to make user an applicant... :hourglass:",
+            ephemeral=True,
+        )
+
+        AUDIT_MESSAGE: Final[str] = f"{ctx.user} used TeX Bot Command \"Make User Applicant\""
+
+        await applicant_member.add_roles(applicant_role, reason=AUDIT_MESSAGE)
+
+        logger.debug("Applicant role given to user %s", applicant_member)
+
+        if guest_role in applicant_member.roles:
+            await applicant_member.remove_roles(guest_role, reason=AUDIT_MESSAGE)
+            logger.debug("Removed Guest role from user %s", applicant_member)
+
+
+        tex_emoji: discord.Emoji | None = self.bot.get_emoji(743218410409820213)
+        if not tex_emoji:
+            tex_emoji = discord.utils.get(main_guild.emojis, name="TeX")
+
+        if intro_channel:
+            recent_message: discord.Message
+            for recent_message in await intro_channel.history(limit=30).flatten():
+                if recent_message.author.id == applicant_member.id:
+                    forbidden_error: discord.Forbidden
+                    try:
+                        if tex_emoji:
+                            await recent_message.add_reaction(tex_emoji)
+                        await recent_message.add_reaction("👋")
+                    except discord.Forbidden as forbidden_error:
+                        if "90001" not in str(forbidden_error):
+                            raise forbidden_error from forbidden_error
+
+                        logger.info(
+                            "Failed to add reactions because the user, %s, "
+                            "has blocked the bot.",
+                            recent_message.author,
+                        )
+                    break
+
+        await initial_response.edit(content=":white_check_mark: User is now an applicant.")
+
+
+class MakeApplicantSlashCommandCog(BaseMakeApplicantCog):
+    """Cog class that defines the "/make_applicant" slash command."""
+
+    @staticmethod
+    async def autocomplete_get_members(ctx: TeXBotApplicationContext) -> set[discord.OptionChoice]: # noqa: E501
+        """
+        Autocomplete callable that generates the set of available selectable members.
+
+        This list of selectable members is used in any of the "make_applicant" slash-command
+        options that have a member input-type.
+        """
+        try:
+            guild: discord.Guild = ctx.bot.main_guild
+            applicant_role: discord.Role = await ctx.bot.applicant_role
+        except (GuildDoesNotExistError, ApplicantRoleDoesNotExistError):
+            return set()
+
+        members: set[discord.Member] = {member for member in guild.members if not member.bot and applicant_role not in member.roles}  # noqa: E501
```

Comment:
Split set comprehension onto multiple lines (rather than ignoring line-length warning):

```python
members: set[discord.Member] = {
    member
    for member
    in guild.members
    if not member.bot and applicant_role not in member.roles
}
```